### ISSUE TITLE
Updates to observables in `firstUpdated` not working

### ISSUE THREAD

**Issue by cannoneyed** [adobe/lit-mobx#26]
I haven't had a chance to dig into this too much, but I'm noticing that setting an observable that the render method uses in `firstUpdated` doesn't trigger a rerender. Setting a LitElement property does, as expected. Here's a simple example of what I mean:

```typescript
// Doesn't work - render isn't called again
class Component extends MobxLitElement {
  @observable test = 1;

  firstUpdated() {
    this.test = 2;
  }

  render() {
    return html`<h1>${this.test}</h1>`;
  }
}
```

```typescript
// Works as expected
class Component extends MobxLitElement {
  @property({type: Number}) test = 1;

  firstUpdated() {
    this.test = 2;
  }

  render() {
    return html`<h1>${this.test}</h1>`;
  }
}
```

I'll try poking around in the source a bit more to learn a bit about what's going on under the hood, but I wanted to open the issue first.

**Comment by benjamind** [comment]
Interesting, please let me know how your investigations pan out and I'll try and take a deeper look at this asap.

**Comment by benjamind** [comment]
Have produced a quick repro:

https://stackblitz.com/edit/typescript-zpfftr

**Comment by benjamind** [comment]
Ok what appears to be happening is that the observable tracking from the `Reaction` setup is not completed until after `firstUpdated` is called, this is due to `firstUpdated` being called synchronously after the render completes.

The `Reaction` we setup is observing the access through the `performUpdate` function which in turn triggers `render`, but the reaction captures the state after this function completes. So making a change in `firstUpdated` is not detected because its already set to the new value as far as the Reaction is concerned..

Not exactly sure the best approach here to fix. @justinfagnani any suggestions? I feel like we 'could' `requestUpdate` after track to force a second render pass to capture any changes after `firstUpdate` but this seems less than ideal in the majority of cases where state is not modified in `firstUpdate`. However we're clearly violating the description of `firstUpdated` right now so we do need to fix this one somehow I think?

**Comment by benjamind** [comment]
One approach that appears to work is to create the Reaction over the `update` method rather than `performUpdate`, this way the reaction is setup over the state before `firstUpdated` is called and we get a requestUpdate reaction after firstUpdated completes. @justinfagnani do you see any issues with this?

**Comment by justinfagnani** [comment]
I suspect this will also be a problem for state changes in `updated()`, and in `update()` after the `super.update()` call.

**Comment by benjamind** [comment]
I guess we could setup separate reactions around each of these functions in a similar manner? Its a little more code, but not a lot, and I guess would solve in each case.

**Comment by justinfagnani** [comment]
Tracking `super.update()` might be ok. Then any subclasses that make state changes after their own `super.update()` call should work.

**Comment by benjamind** [comment]
@cannoneyed this fix has been published under `0.0.4`.

**Comment by cannoneyed** [comment]
Amazing! Thanks so much, might be the fastest fix I've ever seen!
